diff --git a/packages/dev/eslint-config/index.js b/packages/dev/eslint-config/index.js
index facbd26a9..9f6104677 100644
--- a/packages/dev/eslint-config/index.js
+++ b/packages/dev/eslint-config/index.js
@@ -56,7 +56,19 @@ module.exports = {
         '@atlaspack/internal/no-ff-module-level-eval': 'error',
         '@atlaspack/internal/no-relative-import': 'error',
         'flowtype/no-types-missing-file-annotation': 'off',
+        // Temporary
         'prefer-const': 'off',
+        'no-var': 'off',
+        '@typescript-eslint/no-explicit-any': 'off',
+        '@typescript-eslint/no-unused-vars': 'off',
+        '@typescript-eslint/no-require-imports': 'off',
+        '@typescript-eslint/no-misused-new': 'off',
+        '@typescript-eslint/prefer-as-const': 'off',
+        '@typescript-eslint/no-empty-object-type': 'off',
+        '@typescript-eslint/no-unused-expressions': 'off',
+        '@typescript-eslint/no-this-alias': 'off',
+        'no-prototype-builtins': 'off',
+        'prefer-rest-params': 'off',
       },
     },
     {
