diff --git a/node_modules/mocha/lib/reporters/base.js b/node_modules/mocha/lib/reporters/base.js
index 1f97e31..8b86b65 100644
--- a/node_modules/mocha/lib/reporters/base.js
+++ b/node_modules/mocha/lib/reporters/base.js
@@ -379,6 +379,9 @@ Base.prototype.epilogue = function() {
   }
 
   Base.consoleLog();
+  setTimeout(() => {
+    require('why-is-node-running').default();
+  }, 1000);
 };
 
 /**
diff --git a/node_modules/mocha/mocha.js b/node_modules/mocha/mocha.js
index 53ac33e..4243a15 100644
--- a/node_modules/mocha/mocha.js
+++ b/node_modules/mocha/mocha.js
@@ -24745,7 +24745,7 @@
 	    var fmt;
 	    Base.consoleLog(); // passes
 
-	    fmt = color('bright pass', ' ') + color('green', ' %d passing') + color('light', ' (%s)');
+	    fmt = color('bright pass', ' ') + color('green', ' %d passinzzzg') + color('light', ' (%s)');
 	    Base.consoleLog(fmt, stats.passes || 0, ms(stats.duration)); // pending
 
 	    if (stats.pending) {
@@ -24762,6 +24762,8 @@
 	    }
 
 	    Base.consoleLog();
+		Base.consoleLog("hello");
+		require('why-is-node-running').default();
 	  };
 	  /**
 	   * Pads the given `str` to `len`.
